src/file.cr

```diff
@@ -237,25 +240,61 @@ class File < IO::FileDescriptor
     basename(path).chomp(suffix)
   end
 
+  private alias User = System::User | String | Int32 | UInt32
+  private alias Group = System::Group | String | Int32 | UInt32
+
   # Changes the owner of the specified file.
   #
   # ```
-  # File.chown("/foo/bar/baz.cr", 1001, 100)
-  # File.chown("/foo/bar", gid: 100)
+  # File.chown("/foo/bar/baz.cr", "a_owner", "a_group")
+  # File.chown("/foo/bar", group: "a_group")
   # ```
   #
   # Unless *follow_symlinks* is set to `true`, then the owner symlink itself will
   # be changed, otherwise the owner of the symlink destination file will be
   # changed. For example, assuming symlinks as `foo -> bar -> baz`:
   #
   # ```
-  # File.chown("foo", gid: 100)                        # changes foo's gid
-  # File.chown("foo", gid: 100, follow_symlinks: true) # changes baz's gid
+  # File.chown("foo", group: "a_group")                        # changes foo's group to "a_group"
+  # File.chown("baz", group: "a_group", follow_symlinks: true) # changes baz's group to "a_group"
   # ```
-  def self.chown(path, uid : Int = -1, gid : Int = -1, follow_symlinks = false)
+  #
+  # Raises on error.
+  def self.chown(path : String, owner : User? = nil, group : Group? = nil, follow_symlinks : Bool = false) : Nil
+    Crystal::System::File.chown(path, uid_from_user(owner), gid_from_group(group), follow_symlinks)
+  end
+
+  # :nodoc:
+  def self.chown(path : String, uid : Int = -1, gid : Int = -1, follow_symlinks : Bool = false) : Nil
     Crystal::System::File.chown(path, uid, gid, follow_symlinks)
   end
 
+  # Same as `chown()` but instead returns a `Bool` indicating success.
+  # ```
+  def self.chown?(path : String, owner : User? = nil, group : Group? = nil, follow_symlinks : Bool = false) : Bool
+    Crystal::System::File.chown?(path, uid_from_user(owner), gid_from_group(group), follow_symlinks)
+  end
+
+  # :nodoc:
+  private def self.uid_from_user(user : User?) : Int32 | UInt32
```

Comment:
It's probably better to do this with multiple overloads, i.e.

```
private def self.uid_from_user(user : System::User)
  user.uid
end

# ...
```

That way you get exhaustiveness checking for free.